grails-app/services/com/unifina/service/StreamService.groovy

```diff
@@ -328,4 +328,24 @@ class StreamService {
 		reader.readCanvasId()
 		return reader.readModuleId()
 	}
+
+	static class StreamStatus {
+		Boolean ok
+		Date date
+		StreamStatus(Boolean ok, Date date) {
+			this.ok = ok
+			this.date = date
+		}
+	}
+
+	@CompileStatic
+	StreamStatus status(Stream s, Date threshold) {
+		StreamrMessage msg = cassandraService.getLatestFromAllPartitions(s)
+		if (msg == null) {
+			return new StreamStatus(false, null)
+		} else if (msg != null && msg.getTimestamp().before(threshold)) {
```

Comment:
Condition `msg != null` is not strictly necessary here. It will always be true because of the if-condition before.